.github/workflows/test-cloud-platforms.yml

```diff
@@ -0,0 +1,149 @@
+name: Test cloud platforms
+env:
+  REGISTRY: ghcr.io
+  CONTAINER_REPO: ${{ github.repository }}
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+on:
+  schedule:
+    # Each week day at 9:30 UTC (so 3:30 MDT and 2:30 MST).
+    - cron: "30 9 * * MON-FRI"
+
+# Jobs are given a level in a comment.
+# Jobs of the same level run in parallel.
+# Jobs of level N depend of, at least, one job on level N - 1 expect job whom
+# level is 0.
+jobs:
+  check-secrets:
+    name: Check repo secrets
+    # level: 0
+    runs-on: ubuntu-latest
+    outputs:
+      aks: ${{ steps.set_output.outputs.aks }}
+    steps:
+      # Secrets cannot be used as if condition, use job output as workaround.
+      # https://github.com/actions/runner/issues/520
+      - id: set_output
+        run: |
+          if [[ "${{ secrets.AZURE_AKS_CREDS }}" != "" && \
+                "${{ secrets.AZURE_AKS_RESOURCE_GROUP }}" != "" && \
+                "${{ secrets.AZURE_AKS_CLUSTER_PREFIX }}" != "" ]]; \
+          then
+            echo "Secrets to use an ARO cluster were configured in the repo"
+            echo "::set-output name=aks::true"
+          else
+            echo "Secrets to use an ARO cluster were not configured in the repo"
+            echo "::set-output name=aks::false"
+          fi
+
+  build-kubectl-gadget:
+    name: Build kubectl-gadget
+    # level: 0
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [linux, darwin, windows]
+        arch: [amd64, arm64]
+        exclude:
+          - os: windows
+            arch: arm64
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: 'main'
+      - name: Setup go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18.3
+          cache: true
+        id: go
+      - name: Install debian packages
+        uses: ./.github/actions/install-debian-packages
+      - name: Build kubectl-gadget-${{ matrix.os }}-${{ matrix.arch }}
+        run: |
+          git checkout
+
+          # Prevent releases with -dirty suffix due to forgotten entries in
+          # .gitignore.
+          changes="$(git status --porcelain)"
+          if [ -n "$changes" ] ; then
+            echo "$changes"
+            exit 1
+          fi
+
+          kubectl_gadget=kubectl-gadget-${{ matrix.os }}-${{ matrix.arch }}
+
+          CONTAINER_REPO=${{ env.REGISTRY }}/${{ env.CONTAINER_REPO }} \
+          IMAGE_TAG=latest \
+          make $kubectl_gadget
+
+          # Prepare assets for release and actions artifacts
+          platform=$(echo ${kubectl_gadget} | cut -d- -f3-4)
+          mkdir $platform
+          cp $kubectl_gadget $platform/kubectl-gadget
+          cp LICENSE $platform/
+          tar --sort=name --owner=root:0 --group=root:0 \
+            -czf ${kubectl_gadget}.tar.gz -C $platform \
+            kubectl-gadget LICENSE
+          rm -rf $platform
```

Comment:
Is this part necessary in this case? I think we could simply upload the kubectl-gadget binary, isn't it?